src/main/scala/tscfg/generators/java/JavaGen.scala

```diff
@@ -144,16 +151,34 @@ class JavaGen(genOpts: GenOpts) extends Generator(genOpts) {
       ""
     )
 
-    val classStr = {
-      s"""public ${if (isRoot) "" else "static "}class $classNameAdjusted {
-         |  $classDeclMembersStr$classMemberGettersStr
-         |  $membersStr
-         |  public $classNameAdjusted($ctorParams) {
-         |    $errHandlingDecl$ctorMembersStr$errHandlingDispatch
-         |  }$elemAccessorsStr
-         |$rootAuxClasses}
-         |""".stripMargin
-    }
+    val extendsString = abstractClassName.map("extends " + _).getOrElse("")
+    val superString = abstractClassName.map(_ => "super(c, parentPath, $tsCfgValidator);").getOrElse("") // if parent class name is defined a super call is needed
+
+    val classStr =
+      ot match {
+        case _: ObjectType =>
+          s"""public ${
+            if (isRoot) "" else "static "
+          }class $classNameAdjusted $extendsString {
```

Comment:
(1b) and remove it here (as it is adding an extra space right before the `{` for all cases where there's no abstract class involved:
```scala
}class $classNameAdjusted $extendsString{
```